airflow/serialization/serializers/timezone.py

```diff
@@ -17,17 +17,32 @@
 # under the License.
 from __future__ import annotations
 
-from typing import TYPE_CHECKING
+import datetime
+import sys
+from typing import TYPE_CHECKING, cast
 
 from airflow.utils.module_loading import qualname
 
+PY39 = sys.version_info >= (3, 9)
+
 if TYPE_CHECKING:
     from pendulum.tz.timezone import Timezone
 
     from airflow.serialization.serde import U
 
 
-serializers = ["pendulum.tz.timezone.FixedTimezone", "pendulum.tz.timezone.Timezone"]
+serializers = [
+    "pendulum.tz.timezone.FixedTimezone",
+    "pendulum.tz.timezone.Timezone",
+]
+
+if PY39:
+    serializers.append("zoneinfo.ZoneInfo")
+    from zoneinfo import ZoneInfo
+else:
+    serializers.append("backports.zoneinfo.ZoneInfo")
+    from backports.zoneinfo import ZoneInfo
```

Comment:
Technically you can have `backports.zoneinfo` on 3.9+ and I can imagine scenarios where a library installs it for some compatibility edge case. It might not be a bad idea to include the backports serializer unconditionally instead.

I think we should import both `ZoneInfo` classes lazily as well. `import zoneinfo` pulls in timezone data from the system, which can be quite slow depending on the setup.

Reply:
Its an extreme edge case I guess, but I can adjust it